### DISCUSSION TITLE
[Request]: Configurable Pressure plates

### DISCUSSION THREAD

**Discussion by cubesteak** [GriefPrevention/GriefPrevention#189] [Ideas]
### What behaviour is desired:
Feature Request: Allow pressure plates to be configurable.  Say, iron plates only allow trusted players, gold ones allow everyone. Or other way 'round.  :)  I'd like to provide easier use of iron doors with pressure plates, but that isn't doable now.  Allowing configurable pressure plates would allow ease of use for trusted players, but still keep out unwanted players, mobs, etc.

**Comment by RoboMWM** [comment]
You can drop items, have mobs walk on pressure plates, etc. but uh this can also be implemented in an addon, so yea.

**Comment by cubesteak** [comment]
I thought you might say that. :)

Is there a guide to making GP addons?  Not sure that I have the skill to do so, but I'm not against trying.

And btw, TY for taking ownership of this amazing plugin.  Not sure I'd enjoy MC very much without it.

**Comment by RoboMWM** [comment]
Well I'm working on version 20 which will include proper javadocs, but basic gist is basically when a pressure plate is triggered, check location of plate, and cancel if within a claim they don't have access to (or w/e trust perm you wish for it to require).

There's comments in the code but idk if they're sufficient for a beginner. That being said, my first plugin was an addon for GP so maybe  ¯\\\_(ツ)\_/¯

**Comment by cubesteak** [comment]
Hello... Well, I haven't found the time to do the research to learn how to do a GP addon, so wondering if you might be open to making one for a fee?

Not even sure what that might cost, but figured I'd ask you first before posting in the "for hire" jungle on the Spigot forums.  :)

You can reach me on spigotmc.org if you would like to discuss: https://goo.gl/qBSPgQ

Thank you in advance for the kind consideration, totally understand if this isn't something you are interested in.  In that case, please forgive the intrusion.  Also, my prayers are with your Aunt, I hope she has improved.

**Comment by RoboMWM** [comment]
You seem like a nice guy who wouldn't do chargebacks or the sort lol so I could make an exception. However I have no guarantees on time, and I have disabled PMs on spigot due to getting PMs to common support requests at least every week. So if you want to reach me, just hang out in the IRC channel.

Thanks for the prayers, she's okay but weak.

**Comment by cubesteak** [comment]
Will do, TY!  Glad to hear she is OK...
